pkg/executor/mem_reader.go

```diff
@@ -1018,46 +1022,78 @@ func (m *memIndexMergeReader) getMemRowsIter(ctx context.Context) (memRowsIter,
 func (m *memIndexMergeReader) getMemRows(ctx context.Context) ([][]types.Datum, error) {
 	r, ctx := tracing.StartRegionEx(ctx, "memIndexMergeReader.getMemRows")
 	defer r.End()
-	tbls := []table.Table{m.table}
-	// [partNum][indexNum][rangeNum]
-	var kvRanges [][][]kv.KeyRange
-	if m.partitionMode {
-		tbls = tbls[:0]
-		for _, p := range m.partitionTables {
-			tbls = append(tbls, p)
+
+	var err error
+	hMap := kv.NewHandleMap()
+	for i, reader := range m.memReaders {
+		// [partNum][rangeNum]
+		var readerKvRanges [][]kv.KeyRange
+		if m.partitionMode {
+			if r, ok := reader.(*memIndexReader); ok && r.index.Global {
```

Comment:
seems it's part of range build-related logic, can we move this to `buildTableKeyRanges`?

Reply:
and here we can only start with L1044

Reply:
Done